### DISCUSSION TITLE
Standard escapes for space characters don't seem to work

### DISCUSSION THREAD

**Discussion by rcauble** [remkop/picocli#1791] [Q&A]
Hi,

I'm using a modified version of the picocli.shell.jline3.example.Example class in which I've added an Echo command to test string arguments. 

My modification is to add an echo method to the Nested class:

```
        @Command(mixinStandardHelpOptions = true, subcommands = {CommandLine.HelpCommand.class},
                description = "echos a value")
        public void echo(@Option(names = {"--value"}, required = true) String value) {
            System.out.println(value);
        }
```

As long as there's no space is WAI:

```
prompt> cmd nested echo --value foobar
foobar
```

But if I try to add an escaped space character it gets an error:

```
cmd nested echo --value foo\ bar
Unmatched argument at index 5: 'bar'
Usage:  cmd nested echo [-hV] --value=<arg0> [COMMAND]
echos a value
  -h, --help           Show this help message and exit.
  -V, --version        Print version information and exit.
      --value=<arg0>
Commands:
  help  Displays help information about the specified command
```

Upon further digging, I added a println between the reader.readLine and the execute:

```
                        line = reader.readLine(prompt, rightPrompt, (MaskingCallback) null, null);
                        System.out.println(line);
                        systemRegistry.execute(line);
```

When I do that I see

```
cmd nested echo --value foo\ bar
cmd nested echo --value foo bar
Unmatched argument at index 5: 'bar'
Usage:  cmd nested echo [-hV] --value=<arg0> [COMMAND]
echos a value
  -h, --help           Show this help message and exit.
  -V, --version        Print version information and exit.
      --value=<arg0>
Commands:
  help  Displays help information about the specified command
```

So it seems that readLine has removed the escape character. 

So I try double-escaping it:

```
cmd nested echo --value foo\\ bar
cmd nested echo --value foo\ bar
foo\ bar
```

But now echo prints "foo\ bar". 

How do I escape it to make echo print "foo bar"?

**Comment by remkop** [comment]
Hi @rcauble 

One idea: have you tried quoting the parameter that contains the space?

```
prompt> cmd nested echo --value "foo bar"
```

That said, I believe this is a JLine issue, not a picocli issue.
Can you raise this on the JLine issue tracker?

**Reply by rcauble** [reply]
Thanks, I've opened https://github.com/jline/jline3/issues/798

Yes, quoting does solve this particular issue. But I'm trying to find a general purpose rule for how to escape characters for passing strings into jline. For context I have some higher level automation that I'm building on top of a jline application and I was trying to use StringEscapeUtils.escapeXSI to to the argument escaping.

Here's an example of another issue (I mention this in the above link as well) where I'm trying to escape double-quotes and can't seem to find a way to do that:

```
prompt> cmd nested echo --value \"foobar\"
line: cmd nested echo --value "foobar"
foobar
prompt> cmd nested echo --value \\"foobar\\"
line: cmd nested echo --value \"foobar\"
\"foobar\"
prompt> 
```

**Reply by rcauble** [reply]
I found a workaround (I think) and updated it in the bug:

https://github.com/jline/jline3/issues/798#issuecomment-1228016961
